connectors/rabbitmq/src/main/java/io/camunda/connector/rabbitmq/common/model/RabbitMqMessage.java

```diff
@@ -21,39 +21,49 @@
 
 public class RabbitMqMessage {
   private static final Logger LOGGER = LoggerFactory.getLogger(RabbitMqMessage.class);
-  @Secret private Object properties;
-  @NotNull @Secret private Object body;
+  private static final ObjectMapper mapper = new ObjectMapper()
+      .findAndRegisterModules()
+      .configure(DeserializationFeature.FAIL_ON_IGNORED_PROPERTIES, false);
+
+  private Object properties;
+  @NotNull private Object body;
 
   public AMQP.BasicProperties getPropertiesAsAmqpBasicProperties() {
     return Optional.ofNullable(properties)
-        .map(GsonSupplier.gson()::toJsonTree)
+        .map(properties -> mapper.convertValue(properties, JsonNode.class))
         .map(ValidationPropertiesUtil::validateAmqpBasicPropertiesOrThrowException)
         .map(
             jsonProperties ->
-                GsonSupplier.gson().fromJson(jsonProperties, AMQP.BasicProperties.class))
+                mapper.convertValue(jsonProperties, AMQP.BasicProperties.class))
         .orElse(null);
   }
 
   public byte[] getBodyAsByteArray() {
     if (body instanceof String) {
       try {
-        JsonElement jsonElement =
-            GsonSupplier.gson()
-                .fromJson(StringEscapeUtils.unescapeJson(body.toString()), JsonElement.class);
-        if (jsonElement.isJsonPrimitive()) {
+        JsonNode jsonElement =
+            mapper.readTree(StringEscapeUtils.unescapeJson(body.toString()));
```

Comment:
The logic is a bit more complex here, it can be a string or an object, depending on what has been provided by the user.  So it turns out to be not an easy change.